### PR TITLE
docs(data-table): batch actions - update documentation

### LINKED ISSUE #5569: Docs: DataTable Batch Actions Search Bar
## Issue

In the [DataTable documentation](https://github.com/carbon-design-system/carbon/blob/master/packages/react/src/components/DataTable/README.md#batch-actions) for batch actions, the `TableToolbarSearch` component is not listed as a child of `TableToolbarContent`. However, in the running examples in Storybook, it is rendered as a child of `TableToolbarContent`.

image

Not making it a child of `TableToolbarContent` causes incorrect behavior and causes the search button to still render when the select all batch action menu is rendered. If this is indeed a mistake in the docs, I can make a PR fixing it.

### PR DESCRIPTION
**Pull request by andreancardona**
Closes #5569 

Updated data table - Batch Actions documentation. 

#### Changelog

**New**

- Added `TableToolbarSearch` as a child of `TableToolbarContent` in the batch actions documentation and code example and within the filtering code example.

**Changed**

- There should be 3 things updated in this PR 

1.  Updated documentation change to include `TableToolbarSearch` in the Batch Actions - seen here: [DataTable - Batch Actions](https://deploy-preview-6363--carbon-components-react.netlify.app/?path=/story/datatable--with-batch-actions)

2.  Updated Batch Actions code example to show `TableToolbarSearch` nested in `TableToolbarContent` 

3. Updated Filtering code example to show `TableToolbarSearch` nested in `TableToolbarContent` 

#### Testing / Reviewing

- [ ] Please make sure `TableToolbarSearch` is now listed in the [DataTable - Batch Actions](https://deploy-preview-6363--carbon-components-react.netlify.app/?path=/story/datatable--with-batch-actions#batch-actions)

Screen Shot 2020-06-30 at 7 24 49 AM

- [ ] Updated **Batch Actions** code example to show `TableToolbarSearch` nested in `TableToolbarContent` 

Screen Shot 2020-06-30 at 2 32 33 PM

- [ ] Updated **Filtering** code example to show `TableToolbarSearch` nested in `TableToolbarContent` 

Screen Shot 2020-06-30 at 2 31 58 PM

### CODE REVIEW

**Review by joshblack — CHANGES_REQUESTED** [review]
Looking good! Should we also update the code example to reflect this?

image

Seems like we might also need to update this so `TableToolbarSearch` in `TableToolbarContent` 👀 Let me know if I'm missing something!

**Comment by andreancardona** [comment]
> Looking good! Should we also update the code example to reflect this?
> 
> image
> 
> Seems like we might also need to update this so `TableToolbarSearch` in `TableToolbarContent` 👀 Let me know if I'm missing something!

ahh yes of course, I updated the example in filtering as well, lmk how it looks now!

**Review by joshblack — APPROVED** [review]

File: packages/colors/scss/colors.scss [hunk 1]
```diff
@@ -1,12 +0,0 @@
-// Code generated by @carbon/colors. DO NOT EDIT.
```
Comment (tw15egan) [inline packages/colors/scss/colors.scss]: Were these meant to be removed in this PR? 🤔

**Review by tw15egan — APPROVED** [review]
Looks great! 💪 👍 ✅

**Pull request merged**